bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlayFirstTimePopup.java

```diff
@@ -0,0 +1,135 @@
+ * shown, informing the user about the new functionality. This class will track
+ * whether the popup was already shown and will only show the Overlay on the
+ * first time the popup was shown.
+ */
+class FindReplaceOverlayFirstTimePopup {
+
+	private FindReplaceOverlayFirstTimePopup() {
+	}
+
+	private static final String PREFERENCE_NODE_NAME = "org.eclipse.ui.editors"; //$NON-NLS-1$
+	private static final String SETTING_POPUP_WAS_SHOWN_BEFORE = "hasShownOverlayPopupBefore"; //$NON-NLS-1$
+	/**
+	 * How long to wait until the pop up should vanish in Ms.
+	 */
+	private static final Duration POPUP_VANISH_TIME = Duration.ofSeconds(6);
+	private static final String USE_FIND_REPLACE_OVERLAY = "useFindReplaceOverlay"; //$NON-NLS-1$
+
+	/**
+	 * Returns the dialog settings object used to remember whether the popup was
+	 * already shown or not.
+	 *
+	 * @return the dialog settings to be used
+	 */
+	private static IDialogSettings getDialogSettings() {
+		IDialogSettings settings = PlatformUI
+				.getDialogSettingsProvider(FrameworkUtil.getBundle(FindReplaceOverlayFirstTimePopup.class))
+				.getDialogSettings();
+		return settings;
+	}
+
+	private static void disableUseOverlayPreference() {
+		IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(PREFERENCE_NODE_NAME); // $NON-NLS-1$
+		preferences.putBoolean(USE_FIND_REPLACE_OVERLAY, false);
+	}
+
+	/**
+	 * Displays a popup indicating that instead of the FindReplaceDialog, the
+	 * FindReplaceOverlay is currently being used. Only displays the popup on the
+	 * first time use of FindReplaceOverlay.
+	 *
+	 * The popup is bound to the bottom right corner of the principal computer
+	 * Monitor.
+	 *
+	 * @param shellToUse the shell to bind the popup to
+	 */
+	public static void displayPopupIfNotAlreadyShown(Shell shellToUse) {
+		IDialogSettings settings = getDialogSettings();
+
+		if (settings.getBoolean(SETTING_POPUP_WAS_SHOWN_BEFORE) != true) {
+			settings.put(SETTING_POPUP_WAS_SHOWN_BEFORE, true);
+
+			Display displayToUse = Objects.nonNull(shellToUse) ? shellToUse.getDisplay() : Display.getDefault();
+
+			NotificationPopup.forDisplay(displayToUse).content(constructContentCreator())
+					.title(FindReplaceMessages.FindReplaceOverlayFirstTimePopup_FindReplaceOverlayFirstTimePopup_title,
+							true)
+					.delay(POPUP_VANISH_TIME.toMillis()).open();
+		}
+
+	}
+
+	private static Function<Composite, Control> constructContentCreator() {
+		return new Function<>() {
+			@Override
+			public Control apply(Composite t) {
+				return createFirstTimeNotification(t);
+			}
+		};
+	}
+
+	private static Control createFirstTimeNotification(Composite composite) {
+		Link messageBody = new Link(composite, SWT.WRAP);
+
+		messageBody
+				.setText(FindReplaceMessages.FindReplaceOverlayFirstTimePopup_FindReplaceOverlayFirstTimePopup_message);
+		messageBody.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1));
+		messageBody.addSelectionListener(new SelectionListener() {
```

Comment:
Could be replaced by static factory.

Reply:
Yes, fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0a50625bd1074be023a70b085124f92ddcaa83f5